### ISSUE TITLE
nvcc fatal   : Unknown option '-Wall'

### ISSUE THREAD

**Issue by Baboom-l** [traveller59/spconv#289]
running bdist_wheel
running build
running build_py
creating build
creating build/lib.linux-x86_64-3.8
creating build/lib.linux-x86_64-3.8/spconv
copying spconv/functional.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/conv.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/pool.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/__init__.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/test_utils.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/ops.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/tables.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/modules.py -> build/lib.linux-x86_64-3.8/spconv
copying spconv/identity.py -> build/lib.linux-x86_64-3.8/spconv
creating build/lib.linux-x86_64-3.8/spconv/utils
copying spconv/utils/__init__.py -> build/lib.linux-x86_64-3.8/spconv/utils
running build_ext
Release
|||||CMAKE ARGS||||| ['-DCMAKE_PREFIX_PATH=/root/miniconda3/envs/myconda/lib/python3.8/site-packages/torch', '-DPYBIND11_PYTHON_VERSION=3.8', '-DSPCONV_BuildTests=OFF', '-DPYTORCH_VERSION=10500', '-DCMAKE_CUDA_FLAGS="--expt-relaxed-constexpr" -D__CUDA_NO_HALF_OPERATORS__ -D__CUDA_NO_HALF_CONVERSIONS__', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/mnt/spconv/build/lib.linux-x86_64-3.8/spconv', '-DCMAKE_BUILD_TYPE=Release']
-- The CXX compiler identification is GNU 7.5.0
-- The CUDA compiler identification is NVIDIA 10.2.89
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc
-- Check for working CUDA compiler: /usr/local/cuda/bin/nvcc -- works
-- Detecting CUDA compiler ABI info
-- Detecting CUDA compiler ABI info - done
-- Looking for C++ include pthread.h
-- Looking for C++ include pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE  
-- Found CUDA: /usr/local/cuda (found version "10.2") 
-- Caffe2: CUDA detected: 10.2
-- Caffe2: CUDA nvcc is: /usr/local/cuda/bin/nvcc
-- Caffe2: CUDA toolkit directory: /usr/local/cuda
-- Caffe2: Header version is: 10.2
-- Found CUDNN: /usr/lib/x86_64-linux-gnu/libcudnn.so  
-- Found cuDNN: v7.6.5  (include: /usr/include, library: /usr/lib/x86_64-linux-gnu/libcudnn.so)
-- Autodetected CUDA architecture(s):  7.5
-- Added CUDA NVCC flags for: -gencode;arch=compute_75,code=sm_75
-- Found torch: /root/miniconda3/envs/myconda/lib/python3.8/site-packages/torch/lib/libtorch.so  
-- Autodetected CUDA architecture(s):  7.5
-- Found PythonInterp: /root/miniconda3/envs/myconda/bin/python3.8 (found suitable version "3.8.2", minimum required is "3.8") 
-- Found PythonLibs: /root/miniconda3/envs/myconda/lib/libpython3.8.so
-- Performing Test HAS_CPP14_FLAG
-- Performing Test HAS_CPP14_FLAG - Success
-- pybind11 v2.5.0
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Configuring done
-- Generating done
-- Build files have been written to: /mnt/spconv/build/temp.linux-x86_64-3.8
Scanning dependencies of target cuhash
Scanning dependencies of target spconv_nms
[  4%] Building CUDA object src/utils/CMakeFiles/spconv_nms.dir/nms.cu.o
[ 16%] Building CXX object src/cuhash/CMakeFiles/cuhash.dir/hash_table.cpp.o
[ 16%] Building CUDA object src/cuhash/CMakeFiles/cuhash.dir/hash_table.cu.o
[ 16%] Building CUDA object src/cuhash/CMakeFiles/cuhash.dir/hash_functions.cu.o
nvcc fatal   : Unknown option '-Wall'
src/cuhash/CMakeFiles/cuhash.dir/build.make:62: recipe for target 'src/cuhash/CMakeFiles/cuhash.dir/hash_functions.cu.o' failed
make[2]: *** [src/cuhash/CMakeFiles/cuhash.dir/hash_functions.cu.o] Error 1
make[2]: *** Waiting for unfinished jobs....
nvcc fatal   : Unknown option '-Wall'
src/cuhash/CMakeFiles/cuhash.dir/build.make:88: recipe for target 'src/cuhash/CMakeFiles/cuhash.dir/hash_table.cu.o' failed
make[2]: *** [src/cuhash/CMakeFiles/cuhash.dir/hash_table.cu.o] Error 1
In file included from /mnt/spconv/include/cuhash/hash_table.h:21:0,
                 from /mnt/spconv/src/cuhash/hash_table.cpp:18:
/mnt/spconv/include/cuhash/hash_functions.h: In function ‘unsigned int cuhash::stash_hash_function(uint2, unsigned int)’:
/mnt/spconv/include/cuhash/hash_functions.h:84:35: warning: suggest parentheses around arithmetic in operand of ‘^’ [-Wparentheses]
   return (stash_constants.x ^ key + stash_constants.y) % kStashSize;
                               ~~~~^~~~~~~~~~~~~~~~~~~
CMakeFiles/Makefile2:158: recipe for target 'src/cuhash/CMakeFiles/cuhash.dir/all' failed
make[1]: *** [src/cuhash/CMakeFiles/cuhash.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 20%] Linking CUDA static library libspconv_nms.a
[ 20%] Built target spconv_nms
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
Traceback (most recent call last):
  File "setup.py", line 96, in <module>
    setup(
  File "/root/miniconda3/envs/myconda/lib/python3.8/site-packages/setuptools/__init__.py", line 144, in setup
    return distutils.core.setup(**attrs)
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/root/miniconda3/envs/myconda/lib/python3.8/site-packages/wheel/bdist_wheel.py", line 223, in run
    self.run_command('build')
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/root/miniconda3/envs/myconda/lib/python3.8/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "setup.py", line 48, in run
    self.build_extension(ext)
  File "setup.py", line 92, in build_extension
    subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
  File "/root/miniconda3/envs/myconda/lib/python3.8/subprocess.py", line 364, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j4']' returned non-zero exit status 2.

**Comment by rancheng** [comment]
this error is written at the README.md, pytorch 1.4-1.7 has that legacy issue of `-Wall` problem. Luckily, when I test tonight, it seems the problem was fixed by pytorch `1.8`. please install `cuda 11.1` and `cudnn 8.0.5`

**Comment by xavidzo** [comment]
Hello @rancheng, I have this problem with cmake, maybe can you help me please? I am using the latest pytorch 1.8

running bdist_wheel
running build
running build_py
copying spconv/conv.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/functional.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/modules.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/ops.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/pool.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/test_utils.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/__init__.py -> build/lib.linux-x86_64-3.6/spconv
copying spconv/utils/__init__.py -> build/lib.linux-x86_64-3.6/spconv/utils
running build_ext
Release
|||||CMAKE ARGS||||| ['-DCMAKE_PREFIX_PATH=/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch', '-DPYBIND11_PYTHON_VERSION=3.6', '-DSPCONV_BuildTests=OFF', '-DCMAKE_CUDA_FLAGS="--expt-relaxed-constexpr"', '-DCMAKE_LIBRARY_OUTPUT_DIRECTORY=/media/hdd_4tb/xavier/CenterPoint/spconv/build/lib.linux-x86_64-3.6/spconv', '-DCMAKE_BUILD_TYPE=Release']
-- Caffe2: CUDA detected: 11.1
-- Caffe2: CUDA nvcc is: /usr/local/cuda-11.1/bin/nvcc
-- Caffe2: CUDA toolkit directory: /usr/local/cuda-11.1
-- Caffe2: Header version is: 11.1
-- Found cuDNN: v8.0.5  (include: /usr/local/cuda/include, library: /usr/local/cuda/lib64/libcudnn.so)
-- /usr/local/cuda-11.1/lib64/libnvrtc.so shorthash is 3a20f2b6
-- Autodetected CUDA architecture(s):  7.5
-- Added CUDA NVCC flags for: -gencode;arch=compute_75,code=sm_75
-- pybind11 v2.5.0
-- Configuring done
CMake Warning (dev) in src/spconv/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "spconv".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/spconv/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "spconv".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/utils/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "spconv_nms".
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) in src/utils/CMakeLists.txt:
  Policy CMP0104 is not set: CMAKE_CUDA_ARCHITECTURES now detected for NVCC,
  empty CUDA_ARCHITECTURES not allowed.  Run "cmake --help-policy CMP0104"
  for policy details.  Use the cmake_policy command to set the policy and
  suppress this warning.

  CUDA_ARCHITECTURES is empty for target "spconv_nms".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Generating done
-- Build files have been written to: /media/hdd_4tb/xavier/CenterPoint/spconv/build/temp.linux-x86_64-3.6
Scanning dependencies of target spconv_nms
[  7%] Building CUDA object src/utils/CMakeFiles/spconv_nms.dir/nms.cu.o
Scanning dependencies of target spconv
[ 23%] Building CXX object src/spconv/CMakeFiles/spconv.dir/all.cc.o
[ 23%] Building CXX object src/spconv/CMakeFiles/spconv.dir/indice.cc.o
[ 30%] Building CUDA object src/spconv/CMakeFiles/spconv.dir/indice.cu.o
[ 38%] Linking CUDA static library libspconv_nms.a
[ 38%] Built target spconv_nms
[ 46%] Building CXX object src/spconv/CMakeFiles/spconv.dir/reordering.cc.o
/media/hdd_4tb/xavier/CenterPoint/spconv/src/spconv/all.cc:20:91: error: no matching function for call to ‘torch::jit::RegisterOperators::RegisterOperators(const char [28], <unresolved overloaded function type>)’
     torch::jit::RegisterOperators("spconv::get_indice_pairs_2d", &spconv::getIndicePair<2>)
                                                                                           ^
In file included from /home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/script.h:6:0,
                 from /media/hdd_4tb/xavier/CenterPoint/spconv/include/spconv/pool_ops.h:20,
                 from /media/hdd_4tb/xavier/CenterPoint/spconv/src/spconv/all.cc:16:
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:22:3: note: candidate: torch::jit::RegisterOperators::RegisterOperators(std::vector<c10::optional<torch::jit::Operator> >)
   RegisterOperators(std::vector<c10::optional<Operator>> operators) {
   ^
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:22:3: note:   candidate expects 1 argument, 2 provided
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:17:3: note: candidate: constexpr torch::jit::RegisterOperators::RegisterOperators()
   RegisterOperators() = default;
   ^
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:17:3: note:   candidate expects 0 arguments, 2 provided
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:16:18: note: candidate: constexpr torch::jit::RegisterOperators::RegisterOperators(const torch::jit::RegisterOperators&)
 struct TORCH_API RegisterOperators {
                  ^
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:16:18: note:   candidate expects 1 argument, 2 provided
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:16:18: note: candidate: constexpr torch::jit::RegisterOperators::RegisterOperators(torch::jit::RegisterOperators&&)
/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/torch/include/torch/csrc/jit/runtime/custom_operator.h:16:18: note:   candidate expects 1 argument, 2 provided
src/spconv/CMakeFiles/spconv.dir/build.make:81: recipe for target 'src/spconv/CMakeFiles/spconv.dir/all.cc.o' failed
make[2]: *** [src/spconv/CMakeFiles/spconv.dir/all.cc.o] Error 1
make[2]: *** Waiting for unfinished jobs....
Scanning dependencies of target spconv_utils
[ 53%] Building CXX object src/utils/CMakeFiles/spconv_utils.dir/all.cc.o
CMakeFiles/Makefile2:156: recipe for target 'src/spconv/CMakeFiles/spconv.dir/all' failed
make[1]: *** [src/spconv/CMakeFiles/spconv.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 61%] Linking CXX shared library ../../../lib.linux-x86_64-3.6/spconv/spconv_utils.cpython-36m-x86_64-linux-gnu.so
[ 61%] Built target spconv_utils
Makefile:148: recipe for target 'all' failed
make: *** [all] Error 2
Traceback (most recent call last):
  File "setup.py", line 89, in <module>
    zip_safe=False,
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/setuptools/__init__.py", line 153, in setup
    return distutils.core.setup(**attrs)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/site-packages/wheel/bdist_wheel.py", line 299, in run
    self.run_command('build')
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/command/build.py", line 135, in run
    self.run_command(cmd_name)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 40, in run
    self.build_extension(ext)
  File "setup.py", line 73, in build_extension
    subprocess.check_call(['cmake', '--build', '.'] + build_args, cwd=self.build_temp)
  File "/home/xavier/anaconda3/envs/centerpoint_bis/lib/python3.6/subprocess.py", line 311, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j4']' returned non-zero exit status 2.

**Comment by NeoLc** [comment]
> this error is written at the README.md, pytorch 1.4-1.7 has that legacy issue of `-Wall` problem. Luckily, when I test tonight, it seems the problem was fixed by pytorch `1.8`. please install `cuda 11.1` and `cudnn 8.0.5`

solved my problem，thanks！
